src/JWT/Serializers/JsonSerializerFactory.cs

```diff
@@ -1,16 +1,28 @@
 ﻿namespace JWT.Serializers
 {
-    internal static class JsonSerializerFactory
+    public static class JsonSerializerFactory
     {
-        public static IJsonSerializer CreateSerializer()
+        private static IJsonSerializer _serializer;
+
+        public static IJsonSerializer Serializer
         {
-#if SYSTEM_TEXT_JSON
-            return new SystemTextSerializer();
-#elif NEWTONSOFT_JSON
-            return new JsonNetSerializer();
+            get
+            {

```

Comment:
I know why it always returns the same serializer. Because of this line, it's the compile-time condition. Tests that target .NET 4.0 are compiled against higher version but references lower in the runtime.

What we should do instead is to:
1. Provide a parameter in the ctor, e.g. `modernSerializer`
and/or
2. Provide two factories, each would always return its own serializer

Reply:
I don't follow, I thought the compile constant was set to either value depending if it's below or above net462.

Reply:
I mean the idea is to ship below net462 with only json.net, and above with both. Correct?

Reply:
Yeah, that's the idea so that's why it just always imports the json.net nuget and an conditional for the systemtext.json

Reply:
I think because of the combination of these two:

https://github.com/jwt-dotnet/jwt/blob/b38709443e6b96e5757f34b0f91beab869e03bcb/tests/JWT.Tests.Net40/JWT.Tests.Net40.csproj#L4

https://github.com/jwt-dotnet/jwt/blob/b38709443e6b96e5757f34b0f91beab869e03bcb/tests/JWT.Tests.Net40/JWT.Tests.Net40.csproj#L9

So in the build time it targets `net462` and in the runtime - `net40`.

That was a sort of a hack made by a contributor to make very old versions of .NET compilable in the modern environment.

Reply:
I got the test working with net461 so it did detect JSON.net as its default serializer. If I lowered the dotnet version any lower I got errors that other nuggets required a higher dotnet version.